### ISSUE TITLE
[BUG] run_command doesn't seem to report changes

### ISSUE THREAD

**Issue by devicenull** [mborsetti/webchanges#23]
**Describe the bug**
I have run_command configured this way:

```
  run_command:
    enabled: true
    command: /bin/bash -c "echo {count} {jobs} $WEBCHANGES_REPORT_CONFIG_JSON \$WEBCHANGES_REPORT_CONFIG_JSON >>  webchanges.log"
````

My jobs.xml looks like this:

```
name: "meetings"
url: "https://example.com"
---
name: "minutes"
url: "https://example.com"
```

This is on a debian linux box - maybe it only works well on windows?

Am I doing this the wrong way?  I'd like to kick off some other jobs when changes happen, but I need to know which jobs ended up changing.

**Version info**
2022-01-23 17:36:48,670 cli DEBUG: webchanges: 3.8.3 Copyright 2020- Mike Borsetti
2022-01-23 17:36:48,670 cli DEBUG: CPython: 3.9.2 ('default', 'Feb 28 2021 17:03:44') GCC 10.2.1 20210110
2022-01-23 17:36:48,674 cli DEBUG: System: Linux-5.10.0-8-amd64-x86_64-with-glibc2.31

**Comment by devicenull** [comment]
Based on https://github.com/mborsetti/webchanges/blob/45f57ec4dc02daa5aa7be413c5e7f7431f299e59/webchanges/reporters.py#L1586 - it seems like both of these values are hardcoded to 0 - any suggestions on how to get a list of changed jobs?

**Comment by mborsetti** [comment]
Brian,

Thanks and apologies!  You did find the culprit line.

I am about to roll out a new release (testing now) and will include the bugfix; in the interim you can replace that one line with the following:
```
        filtered_job_states = list(self.report.get_filtered_job_states(self.job_states))
        subject_args = {
            'text': text,
            'count': len(filtered_job_states),
            'jobs': ', '.join(job_state.job.pretty_name() for job_state in filtered_job_states ),
        }
        command = shlex.split(self.config['command'].format(**subject_args))
```

**Comment by devicenull** [comment]
Great!  I had also modified my copy locally to have this starting at line 1582 in repoters.py:

```
		changed_jobs = []
		for job_state in self.report.get_filtered_job_states(self.job_states):
			changed_jobs.append({
				'url': job_state.job.get_location(),
				'name': job_state.job.pretty_name(),
			})

		# Work on a copy to not modify the outside environment
		env = dict(os.environ)
		env.update({f'{__project_name__.upper()}_REPORT_CONFIG_JSON': json.dumps(self.report.config)})
		env.update({f'{__project_name__.upper()}_REPORT_CHANGED_JOBS': json.dumps(changed_jobs)})
```

I'm not sure how helpful this is for anyone else - but for me having the url/name of the job in an environment variable saves me having to re-parse the jobs yaml.

**Comment by mborsetti** [comment]
We think alike!!  I already added something very similar to the code:

```
        filtered_job_states = list(self.report.get_filtered_job_states(self.job_states))

        # Work on a copy to not modify the outside environment
        env = dict(os.environ)
        env.update({f'{__project_name__.upper()}_REPORT_CONFIG_JSON': json.dumps(self.report.config)})
        env.update({f'{__project_name__.upper()}_REPORT_REPORTED_JOBS_JSON': json.dumps(self.report.config)})

        subject_args = {
            'text': text,
            'count': len(filtered_job_states),
            'jobs': ', '.join(job_state.job.pretty_name() for job_state in filtered_job_states),
        }
        command = shlex.split(self.config['command'].format(**subject_args))

        try:
            result = subprocess.run(
                command,
                capture_output=True,
                check=True,
                text=True,
                env=env,
            )
```

I am thinking of keeping the report text in the variable {text} (and fixed the code by removing the input=text argument) because there's no need to flood stdin with a potentially very lengthy report even when it's not used by the downstream app.

Let me know if you think your solution is better; I went with dumping the entire job parameters in case there's any need for them downstream.

**Comment by devicenull** [comment]
Dumping the entire job parameters is perfectly fine with me - I had only went with the URL/name because that's all I had needed!

**Comment by mborsetti** [comment]
Fix and additional `WEBCHANGES_REPORT_CHANGED_JOBS` environment variable released in 3.9
